src/SourceBuild/tarball/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/TestScenario.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests
+{
+    public class TestScenario
+    {
+        public DotnetActions Commands { get; }
+        public DotnetLanguage Language { get; }
+        public bool NoHttps { get; set; } = Config.TargetRid.Contains("osx");
+        public string ScenarioName { get; }
+        public DotnetTemplate Template { get; }
+
+        public TestScenario(string scenarioName, DotnetLanguage language, DotnetTemplate template, DotnetActions commands = DotnetActions.None)
+        {
+            ScenarioName = scenarioName;
+            Template = template;
+            Language = language;
+            Commands = commands;
+        }
+
+        internal void Execute(DotNetHelper dotNetHelper)
+        {
+            string templateName = Template.GetName();
+            string languageName = GetLanguageName();
+            string customNewArgs = Template.IsAspNetCore() && NoHttps ? "--no-https" : string.Empty;
+
+            // Can't use '#' in the project name: https://github.com/dotnet/roslyn/issues/51692
+            string projectName = $"{ScenarioName}_{templateName}_{languageName.Replace("#", "Sharp")}";
```

Comment:
Will just `$"{ScenarioName}_{templateName}_{Language}"` here work instead of converting to a different format and then back?

Reply:
Good suggestion - changed.